framework/QuartzRenderer/QuartzRenderer.swift

```diff
@@ -0,0 +1,599 @@
+    public var plotDimensions: PlotDimensions {
+        willSet{
+            if (initialized) {
+                context = CGContext(data: nil,
+                                    width: Int(plotDimensions.frameWidth),
+                                    height: Int(newValue.frameHeight),
+                                    bitsPerComponent: 8,
+                                    bytesPerRow: 0,
+                                    space: colorSpace,
+                                    bitmapInfo: bitmapInfo)!
+                let rect = CGRect(x: 0,
+                                  y: 0,
+                                  width: Int(newValue.frameWidth),
+                                  height: Int(newValue.frameHeight))
+                context.setFillColor(CGColor(red: 1,
+                                             green: 1,
+                                             blue: 1,
+                                             alpha: 1))
+                context.fill(rect)
+            }
+        }
+    }
+
+    public init(width w: Float = 1000, height h: Float = 660, fontPath: String = ""){
+        initialized = false
+        plotDimensions = PlotDimensions(frameWidth: w, frameHeight: h)
+        context = CGContext(data: nil,
+                            width: Int(plotDimensions.frameWidth),
+                            height: Int(plotDimensions.frameHeight),
+                            bitsPerComponent: 8,
+                            bytesPerRow: 0,
+                            space: colorSpace,
+                            bitmapInfo: bitmapInfo)!
+        let rect = CGRect(x: 0,
+                          y: 0,
+                          width: Int(plotDimensions.frameWidth),
+                          height: Int(plotDimensions.frameHeight))
+        context.setFillColor(CGColor(red: 1,
+                                     green: 1,
+                                     blue: 1,
+                                     alpha: 1))
+        context.fill(rect)
+        initialized = true
+    }
+
+    public func drawRect(topLeftPoint p1: Point,
+                         topRightPoint p2: Point,
+                         bottomRightPoint p3: Point,
+                         bottomLeftPoint p4: Point,
+                         strokeWidth thickness: Float,
+                         strokeColor: Color = Color.black,
+                         isOriginShifted: Bool) {
+        let w = abs(p2.x - p1.x)
+        let h = abs(p2.y - p3.y)
+        var y = min(p1.y,p2.y,p3.y,p4.y) + yOffset
+        var x = p1.x + xOffset
+        if (isOriginShifted) {
+            y = y + (0.1*plotDimensions.subHeight)
+            x = x + (0.1*plotDimensions.subWidth)
+        }
+        let rect = CGRect(x: Double(x),
+                          y: Double(y),
+                          width: Double(w),
+                          height: Double(h))
+        context.setStrokeColor(CGColor(red: CGFloat(strokeColor.r),
```

Comment:
I'd recommend creating an extension on Color that returns the equivalent CGColor. It'll save code here and elsewhere, and make things a little easier to read.